tests/client/amo/components/TestAddonReview.js

```diff
@@ -0,0 +1,197 @@
+import React from 'react';
+import { findDOMNode } from 'react-dom';
+import {
+  findRenderedComponentWithType,
+  renderIntoDocument,
+  Simulate,
+} from 'react-addons-test-utils';
+
+import { setReview } from 'amo/actions/reviews';
+import * as amoApi from 'amo/api';
+import * as coreApi from 'core/api';
+import {
+  mapDispatchToProps, mapStateToProps, AddonReviewBase,
+  loadAddonReview as defaultAddonReviewLoader,
+} from 'amo/components/AddonReview';
+import I18nProvider from 'core/i18n/Provider';
+import { fakeAddon, signedInApiState } from 'tests/client/amo/helpers';
+import { getFakeI18nInst, RouterStub } from 'tests/client/helpers';
+
+const defaultReview = {
+  id: 3321, addonSlug: fakeAddon.slug,
+};
+
+function render({ fakeRouter = {}, ...customProps } = {}) {
+  const props = {
+    i18n: getFakeI18nInst(),
+    apiState: signedInApiState,
+    review: defaultReview,
+    updateReviewText: () => {},
+    ...customProps,
+  };
+  const root = findRenderedComponentWithType(renderIntoDocument(
+    <I18nProvider i18n={props.i18n}>
+      <RouterStub router={fakeRouter}>
+        <AddonReviewBase {...props} />
+      </RouterStub>
+    </I18nProvider>
+  ), AddonReviewBase);
+
+  return findDOMNode(root);
+}
+
+describe('AddonReview', () => {
+  it('can update a review', () => {
+    const fakeRouter = {};
+    const updateReviewText = sinon.spy(() => {});
+    const rootNode = render({ updateReviewText, fakeRouter });
+
+    const textarea = rootNode.querySelector('textarea');
+    textarea.value = 'some review';
+    Simulate.submit(rootNode.querySelector('form'));
+
+    assert.equal(updateReviewText.called, true);
+    const params = updateReviewText.firstCall.args[0];
+    assert.equal(params.body, 'some review');
+    assert.equal(params.addonId, defaultReview.addonSlug);
+    assert.equal(params.reviewId, defaultReview.id);
+    assert.equal(params.apiState, signedInApiState);
+    assert.equal(params.router, fakeRouter);
+  });
+
+  it('requires the review text to be non-empty', () => {
+    const rootNode = render();
+    // By default the textarea for the review is empty.
+    try {
+      Simulate.submit(rootNode.querySelector('form'));
+      assert(false, 'unexpected success');
+    } catch (error) {
+      assert.match(error.message, /review .* cannot be empty/);
```

Comment:
I'm surprised the `Simulate.submit()` call raises an exception. Is this because the submit handler is throwing instead of dispatching something to update the state?